### ISSUE TITLE
TOPICS: No way to close long filter dropdowns on mobile except back button, which take you too far back

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1653]
**What is the problem?**

When you open one of the filter dropdowns in which the list is very long (Category, Country and Language), there is no way to close the dropdown without using 'apply'. If you use the back button, this takes you further back than just where you were before opening the filter dropdown, it takes you to the page you were on before (e.g. to the Dashboard).    

![image](https://github.com/user-attachments/assets/196cc055-ea30-49f3-9126-5d1ead9a0b11)

**Why is this a problem?**
Not user-friendly, disorientating.  

**Possible solution.**
I'd suggest there should be a small 'X' to close each of these dropdowns added at the top right of the filter? 

I will add a 'small' label, as I assume it must be :)

**Comment by DiverVM** [comment]
@BeccaMelhuish let's test the solution when input and button stay on the screen, and scrollable just content(languages) area

<img width="272" alt="Image" src="https://github.com/user-attachments/assets/16325989-b753-4149-bf21-587333087d5c" />

**Comment by BeccaMelhuish** [comment]
@DiverVM very good point that it should be like that! Though I'm not sure that will solve the initial problem, with having no way to close it without applying? (I.e. if someone opens it by mistake, or decides not to use it upon opening).

**Comment by ilmartyrk** [comment]
@BeccaMelhuish I agree with @DiverVM, though there might be some confusion the solution will work and I am quite sure people won't get too stuck. I guess other solution would be to set the max-height for the filters are to 80% or even 50% so that the overlay are in the top is visible and clickable. But at the same time it is not that intuitive either

**Comment by DiverVM** [comment]
At least we fixed some styles issues :)
So additionally the solution will be:
- reduce height of the filter to 50-80% as @ilmartyrk proposed
- add additional close(cross) button to the top
- play with UI, for example, switch apply button text from, let's say, close, to apply when filter is choosed
- ....

All options are kinda same effort to implement, just let's choose final option @BeccaMelhuish

**Comment by BeccaMelhuish** [comment]
@DiverVM Middle option of adding additional close(cross) button to the top seems most intuitive to me :)

**Comment by DiverVM** [comment]
@BeccaMelhuish @ilmartyrk added button like this

<img width="230" alt="Image" src="https://github.com/user-attachments/assets/6d07f07b-a685-4716-974e-a62e3687f816" />

**Comment by BeccaMelhuish** [comment]
Seems OK to me, thank you!

**Comment by BeccaMelhuish** [comment]
@DiverVM re-opening as having played around in testing this fix have other thoughts on improvements needed :) 

- [ ] I see now that we actually need a way to close the shorter filters too, and in which case it makes sense to have the same fix across all. So please add the X to all filter pop-ups. 

**A couple of separate issues but makes sense to fix them at the same time I assume? (Leave it if you disagree and I'll make a new issue):** 

For some reason the 'order' filter on discussion arguments is a different format (without tickboxes and apply). Please check all filters have the same tickbox+apply format :)  

Please make sure no two filter pop-up overlay over each other when opened, but that opening a new one closes the first one. E.g. in discussion arguments the 'order' filter currently opens over the top of the 'argument type' filter, if you opened that first.

**Comment by DiverVM** [comment]
@BeccaMelhuish I fix first issue here in this topic, and other 2 I propose to make in separate task, something like "Simplify filters across all places" or "Refactor all filters" or something like this. I see the way we can fix this, it will require to remove some duplicated code, but anyway I would better have a separate task

**Comment by BeccaMelhuish** [comment]
OK wonderful, I will do that - thank you @DiverVM!

**Comment by BeccaMelhuish** [comment]
Tested and all perfect, moving to pre-live :)
